src/CameraSensor.cc

```diff
@@ -319,6 +319,12 @@ bool CameraSensor::Update(const ignition::common::Time &_now)
   // move the camera to the current pose
   this->dataPtr->camera->SetLocalPose(this->Pose());
 
+  // render only if necessary
+  if (!this->dataPtr->pub.HasConnections() &&
+      this->dataPtr->imageEvent.ConnectionCount() <= 0 &&
+      !this->dataPtr->saveImage)
+    return true;
```

Comment:
Is there any way of doing a throttled debug message here?  I feel like this could potentially cause some heartburn when trying to figure out why an image isn't getting where it's supposed to.

Reply:
We've been doing this silently in gazebo-classic and also gazebo_ros_pkgs but I guess more info wouldn't hurt. Instead of throttled debug messages, maybe we could print the msg when the state changes? i.e msg printed when this condition changed from `false` to `true` and vice versa?